server/metadata/tenant.go

```diff
@@ -248,11 +256,11 @@ func (m *TenantManager) GetTenant(ctx context.Context, namespaceName string, txM
 	}
 	id, ok := namespaces[namespaceName]
 	if !ok {
-		return nil, nil
+		return nil, fmt.Errorf("namespace not found: %s", namespaceName)
```

Comment:
do you need to return this from here? Or can the caller handle it? Also, returning like this means not an easy comparison for the caller to detect what type of error it is.